include/openthread/platform/psa.h

```diff
@@ -0,0 +1,237 @@
+#include "psa/crypto_values.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @addtogroup plat-psa
+ *
+ * @brief
+ *   This module includes the platform abstraction for PSA Crypto.
+ *
+ * @{
+ *
+ */
+
+/**
+ * Initialise the PSA module.
+ *
+ * @retval OT_ERROR_NONE          Successfully encrypted  @p aInput.
+ * @retval OT_ERROR_FAILED        Failed to encrypt @p aInput.
+ *
+ */
+otError otPlatPsaInit(void);
+
+/**
+ * Encrypt the given data using AES-ECB cipher.
+ *
+ * @param[in]   aKeyId           Reference to the key to be used for crypto operations.
+ * @param[out]  aInput            A pointer to the input buffer.  Must not be NULL.
+ * @param[in]   aOutput           A pointer to the output buffer.  Must not be NULL.
+ *
+ * @retval OT_ERROR_NONE          Successfully encrypted  @p aInput.
+ * @retval OT_ERROR_FAILED        Failed to encrypt @p aInput.
+ * @retval OT_ERROR_INVALID_ARGS  @p aInput or @p aOutput was set to NULL.
+ *
+ */
+otError otPlatPsaEcbEncrypt(psa_key_id_t aKeyId, const uint8_t *aInput, uint8_t *aOutput);
+
+/**
+ * Generate a key for the parameters specified and store it in PSA ITS.
+ *
+ * @param[out]  aKeyId            Reference to the key to be used for crypto operations.
+ * @param[in]   aKeyType          Key Type encoding for the key.
+ * @param[in]   aKeyAlgorithm     Key algorithm encoding for the key.
+ * @param[in]   aKeyUsage         Key Usage encoding for the key.
+ * @param[in]   aKeyPersistence   Key Persistence for this key
+ * @param[in]   aKeyLen           Length of the key to be generated.
+ *
+ * @retval OT_ERROR_NONE          Successfully encrypted  @p aInput.
+ * @retval OT_ERROR_FAILED        Failed to encrypt @p aInput.
+ * @retval OT_ERROR_INVALID_ARGS  @p aInput or @p aOutput was set to NULL.
+ *
+ */
+otError otPlatPsaGenerateKey(psa_key_id_t *        aKeyId,
+                             psa_key_type_t        aKeyType,
+                             psa_algorithm_t       aKeyAlgorithm,
+                             psa_key_usage_t       aKeyUsage,
+                             psa_key_persistence_t aKeyPersistence,
+                             size_t                aKeyLen);
+
+/**
+ * Import a key into PSA ITS.
+ *
+ * @param[out]  aKeyId            Reference to the key to be used for crypto operations.
+ * @param[in]   aKeyType          Key Type encoding for the key.
+ * @param[in]   aKeyAlgorithm     Key algorithm encoding for the key.
+ * @param[in]   aKeyUsage         Key Usage encoding for the key.
+ * @param[in]   aKeyPersistence   Key Persistence for this key
+ * @param[in]   aKey              Actual key to be imported.
+ * @param[in]   aKeyLen           Length of the key to be imported.
+ *
+ * @retval OT_ERROR_NONE          Successfully encrypted  @p aInput.
+ * @retval OT_ERROR_FAILED        Failed to encrypt @p aInput.
+ * @retval OT_ERROR_INVALID_ARGS  @p aInput or @p aOutput was set to NULL.
+ *
+ */
+otError otPlatPsaImportKey(psa_key_id_t *        aKeyId,
```

Comment:
What is the reason to have otPlatPsaXxxxxKey API? Can't we just use the official PSA API for key handling - https://github.com/ARMmbed/mbedtls/blob/v2.25.0/include/psa/crypto.h#L636

Reply:
The idea was to abstract the PSA Crypto operations entirely and have vendors implement the functionality to fit their design. This would mean that like any platform specific abstractions Crypto operations can shimmed as well.